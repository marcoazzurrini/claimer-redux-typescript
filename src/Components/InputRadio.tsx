/** @jsx jsx */
import { jsx } from "@emotion/core";
import { headingSecondary, headingPrimary } from "../Styles/text";
import { InputRadioList } from "../Styles/inputs";
import PropTypes from "prop-types";
interface Iquestion {
  id: string;
  question: string;
  tag: string;
  answers: [];
  type: string;
}
interface props {
  props: {
    inputVal: string;
    handleChange: Function;
    currentQuestion: Iquestion;
  };
}

export default function InputRadio({ props }: props) {
  const {
    handleChange,
    currentQuestion: { answers, question, type }
  } = props;
  return (
    <ul css={InputRadioList}>
      <p css={headingPrimary}>{question}</p>
      {answers.map(answer => (
        <li key={answer}>
          <input
            value={answer}
            onChange={e => handleChange(e.target.value)}
            type={type}
            required
            id={answer}
            name="option"
          />
          <label css={headingSecondary} htmlFor={answer}>
            {answer}
          </label>
        </li>
      ))}
    </ul>
  );
}

InputRadio.propTypes = {
  optionalObjectWithStrictShape: PropTypes.exact({
    inputVal: PropTypes.string,
    question: PropTypes.object,
    handleChange: PropTypes.func
  })
};
