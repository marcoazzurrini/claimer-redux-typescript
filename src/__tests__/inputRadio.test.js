import React from "react";
import InputRadio from "../Components/InputRadio";
import { shallow, mount } from "enzyme";
import QuizQuestions from "../Api/questions";
let { 0: currentQuestion } = QuizQuestions;
let props = {
  currentQuestion,
  inputVal: "",
  handleChange: () => null
};

describe("testing InputRadio component", () => {
  it("should render without errors", () => {
    shallow(<InputRadio props={props} />);
  });

  it("should match snapshot", () => {
    const wrapper = shallow(<InputRadio props={props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render one to many radio inputs", () => {
    const wrapper = shallow(<InputRadio props={props} />);
    expect(wrapper.find("input")).toBeTruthy();
  });

  it("should render one to many labels", () => {
    const wrapper = shallow(<InputRadio props={props} />);
    expect(wrapper.find("label")).toBeTruthy();
  });

  it("should render one paragraph", () => {
    const wrapper = shallow(<InputRadio props={props} />);
    expect(wrapper.find("paragraph")).toBeTruthy();
  });

  it("should display the question passed in props in the label", () => {
    currentQuestion.question = "mock";
    const wrapper = shallow(<InputRadio props={props} />);
    expect(wrapper.text()).toContain("mock");
    wrapper.unmount();
  });
});
