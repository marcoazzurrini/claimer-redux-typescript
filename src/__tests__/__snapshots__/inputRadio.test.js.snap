// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing InputRadio component should match snapshot 1`] = `ShallowWrapper {}`;
